btree/btree.c

```diff
@@ -3012,21 +3076,23 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 			/* handle if callback fail i.e undo make */
 			node_del(node_slot.s_node, node_slot.s_idx, bop->bo_tx);
 			node_done(&node_slot, bop->bo_tx, true);
-			node_fix(curr_level->l_node, bop->bo_tx);
+			node_fix(lev->l_node, bop->bo_tx);
 			lock_op_unlock(tree);
 			return fail(bop, rc);
 		}
 		node_done(&node_slot, bop->bo_tx, true);
-		node_fix(curr_level->l_node, bop->bo_tx);
+		node_fix(lev->l_node, bop->bo_tx);
 
 		lock_op_unlock(tree);
 		return P_CLEANUP;
 		//return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_DONE);
 	}
-	case P_CLEANUP:
-		level_cleanup(oi);
+	case P_CLEANUP: {
+		oi->i_used = bop->bo_arbor->t_height - 1;
```

Comment:
why this initialization?

Reply:
In some cases , we might be calling P_CLEANUP when oi->i_used is not bop->bo_arbor->t_height - 1 and we are at some internal node, But I can make some changes as well to avoid this initialization